components/org.wso2.carbon.secvault.securevault/src/main/java/org/wso2/carbon/secvault/securevault/SecureVaultInitializer.java

```diff
@@ -18,168 +18,130 @@
-            logger.error("Error while acquiring secure vault configuration", e);
-        }
-    }
-
-    public static synchronized SecureVaultInitializer getInstance() {
-        if (secureVaultInitializer == null) {
-            return new SecureVaultInitializer();
-        }
-        return secureVaultInitializer;
-    }
-
-    /**
-     * Initialise the secure vault for SPI case.
+     * Get secure vault initializer instance.
      *
-     * @param masterKeysFilePath Master key file path which is used store keys and configs to unlock secret repository.
-     * @param secretPropertiesFilePath Secret properties file path which is used to store keys and aliases.
-     * @param secureVaultYAMLPath Secure vault yaml path which is used to store all configuration related to secure
-     *                            vault.
-     * @return SecureVault service instance.
+     * @return secure vault initializer instance
      */
-    public SecureVault initializeSecureVault(String masterKeysFilePath, String secretPropertiesFilePath,
-                                      String secureVaultYAMLPath) {
-        setMasterKeyYAMLPath(Optional.of(masterKeysFilePath));
-        setSecretPropertiesPath(Optional.of(secretPropertiesFilePath));
-        setSecureVaultYAMLPath(Optional.of(secureVaultYAMLPath));
-        initFromSecureVaultYAML();
-        secretRepositoryLoader = ServiceLoader.load(SecretRepository.class);
-        masterKeyReaderLoader = ServiceLoader.load(MasterKeyReader.class);
-        if (masterKeyReaderLoader.iterator().next() == null) {
-            SecureVaultDataHolder.getInstance().setMasterKeyReader(new DefaultMasterKeyReader());
-        } else {
-            SecureVaultDataHolder.getInstance().setMasterKeyReader(masterKeyReaderLoader.iterator().next());
-        }
-        if (secretRepositoryLoader.iterator().next() == null) {
-            SecureVaultDataHolder.getInstance().setSecretRepository(new DefaultSecretRepository());
-        } else {
-            SecureVaultDataHolder.getInstance().setSecretRepository(secretRepositoryLoader.iterator().next());
-        }
-
-        return initializeSecureVault();
+    public static synchronized SecureVaultInitializer getInstance() {
+        return SECURE_VAULT_INITIALIZER;
     }
 
     /**
      * Initialize the secure vault by initialising master key reader and secret repository and loading secrets
      * to secret repository.
+     *
+     * @param secureVaultYAMLPath secure vault YAML path
+     * @return initialized secure vault
+     * @throws SecureVaultException error on initializing secure vault
      */
-    public SecureVault initializeSecureVault() {
+    public SecureVault initializeSecureVault(Path secureVaultYAMLPath) throws SecureVaultException {
         synchronized (this) {
             if (initialized) {
                 logger.debug("Secure Vault Component is already initialized");
                 return new SecureVaultImpl();
```

Comment:
This issue is corrected. SecureVaultInitializer will keep a reference to the secure vault initializer. This is done since we are sure that there can only be a single secure vault instance throughout.